docs/src/components/ComponentDoc/ComponentExample/ComponentExample.tsx

```diff
@@ -203,26 +194,12 @@ class ComponentExample extends React.Component<ComponentExampleProps, ComponentE
     }
   }
 
-  handleShowHTML = (shouldShowHTML: boolean) => {
-    if (shouldShowHTML !== this.state.showHTML) {
-      this.setState({ showHTML: shouldShowHTML }, this.updateHash)
-    }
-  }
-
-  handleShowHTMLInactive = (e: Event) => {
-    if (this.clickedOutsideComponent(e)) {
-      this.handleShowHTML(false)
-    }
-  }
-
   private handleShowRtlClick = e => {
     e.preventDefault()
 
     const { showRtl } = this.state
 
-    this.setState({ showRtl: !showRtl }, () => {
-      this.renderSourceCode()
-    })
+    this.setState({ showRtl: !showRtl })
```

Comment:
would suggest to do it safer, by using state callback arg

Reply:
Agree. I'm planning to refactor this stuff to use a dedicated context, this will simplify `ComponentExample` and `ComponentControls`.

For this PR I want to focus on getting working editor and make changes small as possible :)

Reply:
this will be a small step - just refactor it to the callback and ensure that, before announced changes will be introduced, it will still work correctly. Frankly, don't understand why we cannot eliminate this problem now and improve on mechanism later

Reply:
I refactored it to use `prevState` 👍